Alliance.Common/Extensions/VOIP/Behaviors/VoipHandler.cs

```diff
@@ -266,446 +123,224 @@ public override void OnPreDisplayMissionTick(float dt)
+                CompressVoiceChunk(voiceChunk, ref compressedBuffer, out var compressedBufferLength);
 
-        private int GetPlayerVoiceDataIndex(MissionPeer missionPeer)
-        {
-            for (int i = 0; i < _playerVoiceDataList.Count; i++)
-            {
-                if (_playerVoiceDataList[i].Peer == missionPeer)
-                {
-                    return i;
-                }
+                GameNetwork.BeginModuleEventAsClientUnreliable();
+                GameNetwork.WriteMessage(new SendVoiceRecord(compressedBuffer, compressedBufferLength));
+                GameNetwork.EndModuleEventAsClientUnreliable();
             }
 
-            return -1;
-        }
-
-        private void AddPlayerToVoiceChat(MissionPeer missionPeer)
-        {
-            VirtualPlayer peer = missionPeer.Peer;
-            _playerVoiceDataList.Add(new PeerVoiceData(missionPeer));
-            SoundManager.CreateVoiceEvent();
-            PlatformServices.Instance.CheckPermissionWithUser(Permission.CommunicateUsingVoice, missionPeer.Peer.Id, delegate (bool hasPermission)
+            // Toggle voice recording based on input
+            if (!IsVoiceRecordActive && Mission.InputManager.IsGameKeyPressed(33))
             {
-                if (Mission.Current != null && Mission.Current.CurrentState == Mission.State.Continuing)
-                {
-                    PeerVoiceData playerVoiceData = GetPlayerVoiceData(missionPeer);
-                    if (playerVoiceData != null)
-                    {
-                        if (!hasPermission && missionPeer.Peer.Id.ProvidedType == NetworkMain.GameClient?.PlayerID.ProvidedType)
-                        {
-                            missionPeer.SetMutedFromPlatform(isMuted: true);
-                        }
-
-                        playerVoiceData.SetReadyOnPlatform();
-                    }
-                }
-            });
-            missionPeer.SetMuted(PermaMuteList.IsPlayerMuted(missionPeer.Peer.Id));
-            SoundManager.AddSoundClientWithId((ulong)peer.Index);
-            OnPeerMuteStatusUpdated?.Invoke(missionPeer);
-        }
-
-        private void RemovePlayerFromVoiceChat(int indexInVoiceDataList)
-        {
-            _ = _playerVoiceDataList[indexInVoiceDataList].Peer.Peer;
-            SoundManager.DeleteSoundClientWithId((ulong)_playerVoiceDataList[indexInVoiceDataList].Peer.Peer.Index);
-            SoundManager.DestroyVoiceEvent(indexInVoiceDataList);
-            _playerVoiceDataList.RemoveAt(indexInVoiceDataList);
-
-            // NAudio Cooking
-
-        }
-
-        private void MissionPeerOnTeamChanged(NetworkCommunicator peer, Team previousTeam, Team newTeam)
-        {
-            if (_localUserInitialized && peer.VirtualPlayer.Id != PlayerId.Empty)
+                IsVoiceRecordActive = true;
+            }
+            else if (IsVoiceRecordActive && Mission.InputManager.IsGameKeyReleased(33))
```

Comment:
`33` -> could be useful to know which key represent this `33` on a direct looking